language-support/scala/codegen/src/main/scala/com/digitalasset/codegen/lf/PackageIDsGen.scala

```diff
@@ -30,7 +30,7 @@ object PackageIDsGen {
     val packageIdsSrc: Tree =
       q"""
         package ${Util.packageNameToRefTree(util.packageName)} {
-          private object `Package IDs` {
+          private[${Util.packageNameTailToRefTree(util.packageName)}] object `Package IDs` {
```

Comment:
maybe worth adding a comment that this is just to silence the warning?

Reply:
Honestly this is just more explicit about what the scope is, the unscoped version is kind of a weird corner case in Scala that I still think shouldn't work at all, that we were using just to make the codegen slightly simpler.

Reply:
sounds reasonable, I guess a comment might still save someone from trying to simplify it again but I don’t feel strongly about this.

Reply:
I wish the best of fortune to anyone who would care to simplify any one of dozens of bits of apparent complexity in the codegen and shall soon be acquainted with the blunt force of the ensuing codegen-sample-app compiler errors. 🙂

Reply:
I don't know... we don't document every package private modifier :)
Why should it be different for the generated code.